lib/run/upload-run.js

```diff
@@ -0,0 +1,133 @@
+const util = require('../util');
+
+const _ = require('lodash'),
+print = require('../print'),
+retry = require('async-retry'),
+request = require('postman-request'),
+ERROR_GO_TO_LINK = '<TBA>',
+UPLOAD_RUN_API_URL = '<TBA>'
+
+class uploadRunToPostman{
+
+    constructor(uploadConfig, runSummary){
+        this.uploadConfig = uploadConfig,
+        this.runSummary = runSummary;
+    }
+
+    /**
+     * TBA after discussing with Harsh
+     * @param {*} runSummary
+     * @param {*} collectionUID
+     * @param {*} environmentUID
+     */
+    buildRunObject = ( runSummary, collectionUID , environmentUID ) => {}
+
+    /**
+     * @private
+     * Internal upload call
+     *
+     * @param {Object} uploadOptions
+     * @returns {function} Returns an async function which can be used by async-retry library to have retries
+     */
+     _upload = (uploadOptions) => {
```

Comment:
Why does it need to return a function? Couldn't this function itself be used in `async/retry`?

Reply:
Hint: [.bind()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind)